src/clr-angular/utils/popover/popover-content.ts

```diff
@@ -68,6 +70,18 @@ export class ClrPopoverContent implements AfterContentChecked, OnDestroy {
         } else {
           this.removeContent();
         }
+      }),
```

Comment:
I didn't run any code so I'm not sure if its possible, but, what about combining all three of these observables into a single stream that we can check balues with?

combineLatest(this.smartOpenService.openChange, this.smartPositionService.shouldRealign, this.checkCollector).subscribe([openChange, shouldRealign, checkCollector] => {
  // combine logic for adding content and aligning it here.
})

Reply:
Not really. 
- The CheckCollector is triggered by the ShouldRealign indirectly.
- They don't depend only on OpenChange, but can be triggered manually when manual reposition is needed.
Even if we manage to do it, they are three functionally different observables. I don't see what we could gain in terms of maintainability if we try to minimize the code.